### PR TITLE
[YUNIKORN-569] Display pod namespace/name for the allocations on the apps/nodes page

### PR DESCRIPTION
**Pull request by kobe860219**
### What is this PR for?
Display pod namespace/name for the allocations on the apps/nodes page. Make application-key human readable.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-569

### How should this be tested?
https://github.com/apache/incubator-yunikorn-web/pull/54/checks?check_run_id=2212230697

### Screenshots (if appropriate)
截圖 2021-03-28 下午7 24 42
截圖 2021-03-28 下午7 24 30

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by wilfred-s** [comment]
I think we need to look at the basic display of the data. Yes we need to make thing human readable but we also need to prevent displaying the same information over and over.
* An allocation displayed as part of a node does not need to show the nodeID or the partition. They cannot change for a node and I know which node I am looking at. That information is already there in the overview of the node.
* When looking at an app I see a similar issue. I do not need to see the appID, partition or the queue for the allocation information when I look at an app. That is always the same and should just be shown in the app overview.

Re-ordering of the data that is displayed, i.e. appID should be close the the key when looking at the node allocations. etc will also improve readability a lot.
Last point to make is that removing the key for allocations from the web UI will break any possibility to track that allocation in the logs. The key (ID) will be logged in the files so if you need to troubleshoot anything you need that and we cannot just remove it.

**Comment by kobe860219** [comment]
@yangwwei  @wilfred-s 
image

**Comment by yangwwei** [comment]
@kobe860219  could u pls rebase because #51 just got merged. And pls fix the build issue https://travis-ci.com/github/apache/incubator-yunikorn-web/builds/221802141

**Comment by akhilpb001** [comment]
I agree with the improvements suggested by @wilfred-s. I feel that we can do both those improvements in one jira and no need for two jira.

@kobe860219  Could you pls fix the AllocationKey value "some-namespace/undefined"? If both namespace and podname are not available, pls fallback to display allocationId as we have had earlier.

**Review by akhilpb001 — CHANGES_REQUESTED** [review]
Added comments to fix.

File: src/app/services/scheduler/scheduler.service.ts [hunk 1]
```diff
@@ -86,9 +87,16 @@ export class SchedulerService {
...
             if (allocations && allocations.length > 0) {
               const appAllocations = [];
-              allocations.forEach(alloc => {
+              allocations.forEach((alloc) => {
+                let displayName =
+                  alloc.allocationTags['kubernetes.io/meta/namespace'] +
+                  '/\r' +
+                  alloc.allocationTags['kubernetes.io/meta/podName'];
+                alloc['displayName'] = displayName;
+                console.log(alloc['displayName']);
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:96]: remove console log

File: src/app/services/scheduler/scheduler.service.ts [hunk 2]
```diff
@@ -86,9 +87,16 @@ export class SchedulerService {
             const allocations = app['allocations'];
             if (allocations && allocations.length > 0) {
               const appAllocations = [];
-              allocations.forEach(alloc => {
+              allocations.forEach((alloc) => {
+                let displayName =
+                  alloc.allocationTags['kubernetes.io/meta/namespace'] +
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:92]: use string literal format

File: src/app/services/scheduler/scheduler.service.ts [hunk 3]
```diff
@@ -31,16 +31,17 @@ import { AllocationInfo } from '@app/models/alloc-info.model';
 import { HistoryInfo } from '@app/models/history-info.model';
 import { NodeInfo } from '@app/models/node-info.model';
 import { NOT_AVAILABLE } from '@app/utils/constants';
+import { connectableObservableDescriptor } from 'rxjs/internal/observable/ConnectableObservable';
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:34]: remove this if its not used.

File: src/app/services/scheduler/scheduler.service.ts [hunk 4]
```diff
@@ -73,7 +74,7 @@ export class SchedulerService {
       map((data: any) => {
         const result = [];
         if (data && data.length > 0) {
-          data.forEach(app => {
+          data.forEach((app) => {
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:77]: bracket for single param in arrow function is optional.

File: src/app/services/scheduler/scheduler.service.ts [hunk 5]
```diff
@@ -61,7 +62,7 @@ export class SchedulerService {
         const partitionName = data['partitionname'] || '';
         return {
           rootQueue,
-          partitionName
+          partitionName,
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:65]: trailing comma is not needed, may be added by your editor.

File: src/app/services/scheduler/scheduler.service.ts (hunk 3 above)
Comment (kobe860219) [reply src/app/services/scheduler/scheduler.service.ts:34]: Removed

File: src/app/services/scheduler/scheduler.service.ts (hunk 1 above)
Comment (kobe860219) [reply src/app/services/scheduler/scheduler.service.ts:96]: Removed

**Comment by kobe860219** [comment]
Fix  https://issues.apache.org/jira/browse/YUNIKORN-615 in this pr.

File: src/app/services/scheduler/scheduler.service.ts [hunk 6]
```diff
@@ -87,8 +87,19 @@ export class SchedulerService {
...
               allocations.forEach(alloc => {
+                if (
+                  alloc.allocationTags['kubernetes.io/meta/namespace'] &&
+                  alloc.allocationTags['kubernetes.io/meta/podName']
+                ) {
+                  alloc[
+                    'displayName'
+                  ] = `${alloc.allocationTags['kubernetes.io/meta/namespace']}/\r${alloc.allocationTags['kubernetes.io/meta/podName']}`;
+                } else {
+                  alloc['displayName'] = `some-namespace/undefined`;
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:98]: @kobe860219 It does not make sense to display `some-namespace/undefined` when above specific allocationTags are not available. Could you pls change this to `n/a` may be?

**Review by akhilpb001 — CHANGES_REQUESTED** [review]

File: src/app/services/scheduler/scheduler.service.ts [hunk 7]
```diff
@@ -87,8 +87,19 @@ export class SchedulerService {
             if (allocations && allocations.length > 0) {
               const appAllocations = [];
               allocations.forEach(alloc => {
+                if (
+                  alloc.allocationTags['kubernetes.io/meta/namespace'] &&
```
Comment (akhilpb001) [inline src/app/services/scheduler/scheduler.service.ts:91]: Change this if check to `alloc.allocationTags && alloc.allocationTags['kubernetes.io/meta/namespace'] && alloc.allocationTags['kubernetes.io/meta/podName']`

**Comment by akhilpb001** [comment]
Also pls attach latest screenshots with the fix.

**Comment by kobe860219** [comment]
@akhilpb001  Here is newest screenshots.
截圖 2021-04-14 下午12 59 31
截圖 2021-04-14 下午12 59 42

**Review by akhilpb001 — APPROVED** [review]

**Pull request merged**